### PR TITLE
Add eslint and update webpack 3

### PR DESCRIPTION
**Pull request by le1tuan**
-Install Eslint extends Standard
-Update Webpack: babel loader , webpack-dev-server
-Split webpack config file.
-Double check eslint with pre-commit

### CODE REVIEW

File: webpack.config.js [hunk 1]
```diff
@@ -24,8 +24,20 @@ module.exports = {
...
+      { 
+        test: /\.js?$/,
+        exclude: /node_modules/,
+        use: {
+          loader: 'babel-loader',
+          options: {
+            presets: ["env","react","stage-0",
+            "stage-1",
+            "stage-2"],
+            plugins: ["transform-decorators-legacy"]
```
Comment (codeaholicguy) [inline webpack.config.js]: Using `.babelrc`

File: webpack.config.prod.js [hunk 2]
```diff
@@ -12,14 +12,29 @@ module.exports = {
...
+      { 
+        test: /\.js?$/, 
+        exclude: /node_modules/, 
+        use: {
+          loader: 'babel-loader',
+          options: {
+            presets: ["env","react","stage-0",
+            "stage-1",
+            "stage-2"],
+            plugins: ["transform-decorators-legacy"]
```
Comment (codeaholicguy) [inline webpack.config.prod.js]: Create `webpack.base.config` for preventing duplication in code

File: package.json [hunk 3]
```diff
@@ -4,9 +4,9 @@
   "description": "What is the best programming language for me?",
   "main": "index.js",
   "scripts": {
-    "start":
-      "webpack-dev-server --progress --colors --hot --config ./webpack.config.js",
-    "build": "webpack --progress --colors --config ./webpack.config.prod.js"
+    "start": "webpack-dev-server --progress --colors --hot --config ./webpack.config.js",
+    "build": "webpack --progress --colors --config ./webpack.config.prod.js",
+    "lint:eslint": "./node_modules/.bin/eslint ./src"
```
Comment (codeaholicguy) [inline package.json]: Double check and fix linter error if any

**Comment by codeaholicguy** [comment]
Thanks for your contribution, there are some comments need to be addressed

**Comment by le1tuan** [comment]
@codeaholicguy  Please check again all my changes

**Comment by codeaholicguy** [comment]
@le1tuan Thanks for your time

**Pull request merged**